docs/design/2020-06-01-global-kill.md

```diff
@@ -48,26 +54,55 @@ Bit 63 is always __ZERO__, making `connId` in range of non-negative int64, to be
 -  `markup == 1` indicates that the `connID` is 64 bits long. Incompatible with legacy 32 bits clients.
 -  `markup == 1` while __high 32 bits are zeros__, indicates that 32 bits truncation happens. See `Compatibility` section.
 
-
 #### 4. serverId
-`serverId` is selected RANDOMLY by each TiDB instance on startup, and the uniqueness is guaranteed by PD(etcd). `serverId` should be larger or equal to 1, to ensure that high 32 bits of `connId` is always non-zero, and make it possible to detect truncation.
-
-On failure (e.g. fail connecting to PD, or all `serverId` are unavailable), we block any new connection.
+- `serverId` is selected RANDOMLY from `serverIds pool`_(see next)_ by each TiDB instance on startup, and the uniqueness is guaranteed by PD (etcd). `serverId` should be larger or equal to 1, to ensure that high 32 bits of `connId` is always non-zero, and make it possible to detect truncation.
 
-`serverId` is kept by PD with a lease (defaults to 12 hours, long enough to avoid brutally killing any long-run SQL). If TiDB is disconnected to PD longer than half of the lease (defaults to 6 hours), all connections are killed, and no new connection is accepted, to avoid running with stale/incorrect `serverId`. On connection to PD restored, a new `serverId` is acquired before accepting new connection.
+- `serverIds pool` is:
+  - All __UNUSED__ `serverIds` within [1, 2047] acquired from [`CLUSTER_INFO`](https://docs.pingcap.com/tidb/stable/information-schema-cluster-info#cluster_info) when 32 bits `connId` chosen.
+  - All `serverIds` within [2048, 2^22-1] when 64 bits `connId` chosen.
 
-On single TiDB instance without PD, a `serverId` of `1` is assigned.
+- On failure (e.g. fail connecting to PD, or all `serverId` are unavailable when 64 bits `connId` chosen), we block any new connection.
+- `serverId` is kept by PD with a lease (defaults to 12 hours, long enough to avoid brutally killing any long-run SQL). If TiDB is disconnected to PD longer than half of the lease (defaults to 6 hours), all connections are killed, and no new connection is accepted, to avoid running with stale/incorrect `serverId`. On connection to PD restored, a new `serverId` is acquired before accepting new connection.
+- On single TiDB instance without PD, a `serverId` of `1` is assigned.
 
```

Comment:
Could you illustrate a bit more about how serverId is acquired from the pool? Do they behave the same for both 32bit and 64bit?

Reply:
I add a "summary" section for illustrating more clearly. PTAL.